### PR TITLE
OCD-4246: Support unsubscribing from all CHPL notifications

### PR DESCRIPTION
**Pull request by kekey1**
[#OCD-4246]

### CODE REVIEW

File: src/app/api/subscriptions.jsx [hunk 1]
```diff
@@ -20,8 +20,14 @@ const usePutSubscriber = () => {
...
+const useDeleteSubscriber = () => {
+  const axios = useAxios();
+  return useMutation(async (data) => axios.put('subscriptions/unsubscribe-all', data));
+};
+
 export {
   useFetchRoles,
   usePostSubscription,
   usePutSubscriber,
+  useDeleteSubscriber,
```
Comment (andlar) [inline src/app/api/subscriptions.jsx]: For all of the other `api/*.jsx` files, the export functions are alphabetical, and the related functions are ordered in the file the same way

File: src/app/pages/subscriptions/unsubscribe-all.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const [message, setMessage] = useState(undefined);
+  const deleteSubscriber = useDeleteSubscriber();
+  const classes = useStyles();
+
+  useEffect(() => {
+    deleteSubscriber.mutate({
+      subscriberId: hash,
+    }, {
+      onSuccess: (response) => {
+        console.log({ response });
```
Comment (andlar) [inline src/app/pages/subscriptions/unsubscribe-all.jsx]: Can be removed

File: src/app/pages/subscriptions/unsubscribe-all.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        setMessage('You have been unsubscribed from all CHPL notifications');
+      },
+      onError: (error) => {
+        setMessage(error.response.data.error);
```
Comment (andlar) [inline src/app/pages/subscriptions/unsubscribe-all.jsx:33]: I like the API sending back a useful message, but I wonder if the UI should show something more generic. Kind of like how we don't tell them if it's their username or password that they have wrong. I should have brought it up at the FR, but forgot. Thoughts?

**Review by andlar — DISMISSED** [review]
I have a couple of style notes, and a question we might need to get answered by the PO, but am fine with this going to QA as is

**Review by andlar — APPROVED** [review]

**Review by tmy1313 — APPROVED** [review]

**Pull request merged**